### PR TITLE
Platml 4759

### PR DESCRIPTION
**Pull request by tumulurik**

### CODE REVIEW

File: bootstrap/data_ingester.py [hunk 1]
```diff
@@ -25,53 +25,49 @@
...
-            "persisted": True,
-            "containerFormat": "parquet",
-            "format": "json"
-        },
-        "aspect": "production"
-    }
+
+    # Set the title and description
+    data['name'] = dataset_title
+    data['description'] = dataset_title
```
Comment (rjrussell77) [inline bootstrap/data_ingester.py:41]: Should this be, `dataset_description`?

File: bootstrap/config.yaml [hunk 2]
```diff
@@ -27,15 +32,15 @@ class_data:
...
+input_mixin_data:
   "type": "object"
   "title": "mixin_title"
   "description": "mixin_title"
   "meta:intendedToExtend": [{"class_id"}]
   "definitions": {
     "retail": {
       "properties": {
-        "tenant_id": {
+        "_acpmlexploratoryexternal": {
```
Comment (rjrussell77) [inline bootstrap/config.yaml]: Is this correct?  Thought things were becoming abstract.

File: bootstrap/data_ingester.py (hunk 1 above)
Comment (tumulurik) [reply bootstrap/data_ingester.py:41]: Does not matter for all the inputs and outputs class, schema and mixins using the title as the description to optimize the number of key, value pairs in the config file

File: bootstrap/config.yaml (hunk 2 above)
Comment (tumulurik) [reply bootstrap/config.yaml]: Yes, reverted back

File: bootstrap/bootstrap.py [hunk 3]
```diff
@@ -38,36 +38,42 @@ def ingest():
...
-    priv_key_file = open(priv_key_filename, "r")
-    priv_key = priv_key_file.read()
-    priv_key_file.close()
+    ims_token = cfg['Platform']['ims_token']
 
-    # Get the platform url
-    platform_gateway_url = cfg['Platform']['platform_gateway']
+    if ims_token is None:
+        # Server parameters
+        ims_host = cfg["Server"]["ims_host"]
```
Comment (rjrussell77) [inline bootstrap/bootstrap.py]: I see two quote formats used for cfg, single and double quotes? (e.g. `cfg['Platform']` and `cfg["Server"]`)

File: bootstrap/bootstrap.py [hunk 4]
```diff
@@ -38,36 +38,42 @@ def ingest():
...
+        priv_key_file.close()
+        ims_token = "Bearer " + get_access_token(ims_host, ims_endpoint_jwt, org_id, tech_acct, api_key,
+                                                 client_secret, priv_key)
 
     # Get the titles for the class, mixin, schema and dataset
-    class_title = cfg['Titles for Schema and Dataset']['class_title']
-    mixin_title = cfg['Titles for Schema and Dataset']['mixin_title']
-    schema_title = cfg['Titles for Schema and Dataset']['schema_title']
-    dataset_title = cfg['Titles for Schema and Dataset']['dataset_title']
+    input_class_title = cfg['Titles for Schema and Dataset']['input_class_title']
```
Comment (rjrussell77) [inline bootstrap/bootstrap.py]: Replace literals with constants? (e.g. `'Titles for Schema and Dataset'`)

File: bootstrap/config.yaml [hunk 5]
```diff
@@ -6,18 +6,24 @@ Enterprise:
...
   org_id: 20656D0F5B9975B20A495E23@AdobeOrg
   tech_acct: F98D7E725CCA6F770A495FD9@techacct.adobe.com
-  client_secret: client_secret
-  priv_key_filename: private.key
+  client_secret: <client_secret>
+  priv_key_filename: private_for_test_external.key
 
 Platform:
   platform_gateway: https://platform.adobe.io
+  ims_token:
```
Comment (rjrussell77) [inline bootstrap/config.yaml]: Need a placeholder here?

File: bootstrap/data_ingester.py [hunk 6]
```diff
@@ -25,53 +25,49 @@
...
     """
     Get the batchId by making a POST request to "/data/foundation/import/batches"
     :param create_batch_url: url
     :param headers: headers
     :param dataset_id: dataset id
+    :param data: post request data
     :return: batch id
     """
-    data_for_create_batch = {"datasetId": dataset_id}
+    data["datasetId"] = dataset_id
```
Comment (rjrussell77) [inline bootstrap/data_ingester.py:66]: Would it be normal to check for null/empty `data` param before de-referencing it?

**Review by rjrussell77 — APPROVED** [review]
Some minor things to look at but otherwise, LGTM

**Review by tatsuyawi — APPROVED** [review]

File: bootstrap/schema_ingester.py [hunk 7]
```diff
@@ -75,8 +75,8 @@ def get_mixin_id(create_mixin_url, headers, mixin_title, data, class_id, tenant_
     # Set the class id
     data['meta:intendedToExtend'][0] = class_id
     # Set the tenant id
-    data["definitions"]["retail"]["properties"][str(tenant_id)] = data["definitions"]["retail"]["properties"][
-        "tenant_id"]
+    data["definitions"]["retail"]["properties"][str(tenant_id)] = \
+        data["definitions"]["retail"]["properties"]["tenant_id"]
```
Comment (jonaphin) [inline bootstrap/schema_ingester.py]: I do not think there should be any references to retail in the ingester

**Review by jonaphin — CHANGES_REQUESTED** [review]

**Review by jonaphin — APPROVED** [review]
perfect! Thank you :)

**Pull request merged**